### PR TITLE
Mock server for e2e tests using Prism

### PR DESCRIPTION
**Pull request by develohpanda**
As per title 😄 

Related to #2535

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -43,3 +43,6 @@ jobs:
         run: npm run bootstrap
       - name: Run tests
         run: npm test
+      # - name: Run prism
+      #   uses: ./packages/insomnia-smoke-test/prism # Uses an action in the root directory
+      #   id: prism
```
Comment (develohpanda) [inline .github/workflows/test.yml]: To be added when testing is added to workflows (part of a separate PR)

File: packages/insomnia-smoke-test/package.json [hunk 2]
```diff
@@ -1,4 +1,5 @@
 {
+  "private": true,
```
Comment (develohpanda) [inline packages/insomnia-smoke-test/package.json:2]: Tells `lerna` to not publish to npm

File: packages/insomnia-smoke-test/spec.js [hunk 3]
```diff
@@ -40,56 +40,56 @@ describe('Application launch', function() {
...
+    const firstRequestName = await firstRequest.$('span.editable').then(e => e.getText());
+    const firstRequestClasses = await firstRequest.getAttribute('class');
+
+    expect(firstRequestName).toBe(requestName);
+    expect(firstRequestClasses).toContain('sidebar__item--active');
+
+    // Type into url bar
+    const urlEditor = await app.client.$('.urlbar .editor');
+    await urlEditor.click();
+    await urlEditor.keys('http://127.0.0.1:4010/pets/1');
```
Comment (develohpanda) [inline packages/insomnia-smoke-test/spec.js:89]: Needs to be extracted to a common file, eventually. Not in this PR.

File: packages/insomnia-smoke-test/README.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+# Insomnia Smoke Tests
+
+```sh
+npm install
+
+npm run test:core:build
+npm run test:designer:build
+npm run test:core:package
+npm run test:core:build
```
Comment (DMarby) [inline packages/insomnia-smoke-test/README.md]: Is this meant to be `npm run test:designer:package`?

File: packages/insomnia-smoke-test/spec.js [hunk 5]
```diff
@@ -40,56 +40,56 @@ describe('Application launch', function() {
...
-  //   await app.client
-  //     .$('.response-pane .pane__header .tag.bg-success')
-  //     .then(e => e.getText())
-  //     .then(e => expect(e).toBe('200 OK'));
-  //
-  //   await app.browserWindow.capturePage().then(function(imageBuffer) {
-  //     fs.writeFileSync('page.png', imageBuffer);
-  //   });
-  // });
+  it('create and send a request', async () => {
```
Comment (DMarby) [inline packages/insomnia-smoke-test/spec.js]: Nitpick 
```suggestion
  it('creates and sends a request', async () => {
```

**Review by DMarby — APPROVED** [review]
This is so cool 🎉

File: packages/insomnia-smoke-test/README.md (hunk 4 above)
Comment (develohpanda) [reply packages/insomnia-smoke-test/README.md]:
```suggestion
npm run test:designer:package
```

**Review by DMarby — APPROVED** [review]

**Review by DMarby — APPROVED** [review]

File: package.json [hunk 6]
```diff
@@ -25,7 +25,6 @@
     "storybook-start": "npm run storybook --prefix packages/insomnia-components",
     "storybook-legacy-start": "npm run storybook --prefix packages/insomnia-app",
     "preinstall": "node ./scripts/check-version.js",
-
     "app-build:smoke:core": "cross-env SMOKE_TEST=core npm run app-build",
     "app-build:smoke:designer": "cross-env SMOKE_TEST=designer npm run app-build",
     "test:smoke:core": "npm run test:build:core --prefix packages/insomnia-smoke-test",
```
Comment (DMarby) [inline package.json]: I think this and the ones below needs to be changed to `test:<app>:build`?

```suggestion
    "test:smoke:core": "npm run test:core:build --prefix packages/insomnia-smoke-test",
```

Comment (develohpanda) [reply package.json]: Still working on this naming 😅 thanks

**Pull request merged**